internal/checker/utilities.go

```diff
@@ -1007,6 +1007,23 @@ func getPropertyNameFromType(t *Type) string {
 	panic("Unhandled case in getPropertyNameFromType")
 }
 
+func GetEffectiveImplementsTypeNodes(node *ast.Node) []*ast.TypeNode {
```

Comment:
I think we won't need to port this given the new JSDoc approach is to add synthetic nodes precisely to avoid this kind of function. Same for `GetEffectiveBaseTypeNode` below. You should probably use the existing `GetImplementsTypeNodes`.